src/PSY2PRAS.jl

```diff
@@ -264,73 +253,82 @@ function generate_pras_system(
                 x -> (PSY.get_prime_mover_type(x) == PSY.PrimeMovers.PVe),
                 reg_ren_comps,
             )
-            reg_gen_comps =
-                availability ?
-                get_available_components_in_aggregation_topology(
-                    PSY.Generator,
-                    sys,
-                    region,
-                ) : PSY.get_components_in_aggregation_topology(PSY.Generator, sys, region)
-            gs = filter(
-                x -> (
-                    ~(typeof(x) == PSY.HydroEnergyReservoir) &&
-                    ~(iszero(PSY.get_max_active_power(x))) &&
-                    PSY.IS.get_uuid(x) ∉ union(
-                        s2p_meta.hs_uuids,
-                        PSY.IS.get_uuid.(wind_gs),
-                        PSY.IS.get_uuid.(pv_gs),
+            if (length(wind_gs) > 1)
+                temp_lumped_wind_gen = PSY.RenewableDispatch(nothing)
+                PSY.set_bus!(temp_lumped_wind_gen, PSY.get_bus(first(wind_gs)))
+                PSY.set_name!(temp_lumped_wind_gen, "Lumped_Wind_" * region_names[idx])
+                PSY.set_prime_mover_type!(temp_lumped_wind_gen, PSY.PrimeMovers.WT)
+                ext = PSY.get_ext(temp_lumped_wind_gen)
+                ext["region_gen_ts"] =
+                    round.(
+                        Int,
+                        sum(
+                            get_ts_values.(
+                                get_first_ts.(
+                                    PSY.get_time_series_multiple.(
+                                        wind_gs,
+                                        s2p_meta.filter_func,
+                                        name="max_active_power",
+                                    )
+                                )
+                            ) .* PSY.get_max_active_power.(wind_gs),
+                        ),
                     )
-                ),
-                reg_gen_comps,
-            )
-            push!(gens, gs)
-            push!(reg_wind_gens, wind_gs)
-            push!(reg_pv_gens, pv_gs)
-
-            if (idx == 1)
-                start_id[idx] = 1
-            else
-                if (length(reg_wind_gens[idx - 1]) > 0 && length(reg_pv_gens[idx - 1]) > 0)
-                    start_id[idx] = start_id[idx - 1] + length(gens[idx - 1]) + 2
-                elseif (
-                    length(reg_wind_gens[idx - 1]) > 0 || length(reg_pv_gens[idx - 1]) > 0
-                )
-                    start_id[idx] = start_id[idx - 1] + length(gens[idx - 1]) + 1
-                else
-                    start_id[idx] = start_id[idx - 1] + length(gens[idx - 1])
+                PSY.set_base_power!(temp_lumped_wind_gen, maximum(ext["region_gen_ts"]))
+                PSY.set_rating!(temp_lumped_wind_gen, maximum(ext["region_gen_ts"]))
+                PSY.add_component!(sys, temp_lumped_wind_gen)
+                for comp in wind_gs
+                    PSY.remove_component!(sys, comp)
```

Comment:
Why are we removing this component? Can't we just create the appropriate generator from the new lumped one without having to add it to the system?